data/transactions/logic/eval.go

```diff
@@ -584,7 +587,31 @@ func (cx *evalContext) step() {
 		cx.err = fmt.Errorf("pc=%3d dynamic cost budget of %d exceeded, executing %s", cx.pc, cx.budget(), spec.Name)
 		return
 	}
+
+	preheight := len(cx.stack)
 	spec.op(cx)
+
+	if cx.err == nil {
+		postheight := len(cx.stack)
+		if spec.Name != "return" && postheight-preheight != len(spec.Returns)-len(spec.Args) {
```

Comment:
This error would only occur because of a bug in our implementation, not in teal code.  I don't know how user-friendly it needs to be.  The info *we* need is there:  There's a bug in the named opcode and it didn't manage the stack properly.

Reply:
I simply do not remember checks for popped vs pushed. But I guess you are right and these problem would be already discovered.

Reply:
Yes, that's true, those checks are new.  But they are checks on the results of individual opcode operations, so I don't think it is possible to construct a strange program that would trip them.  If they catch a problem, it would mean that one of our existing opcodes has a pretty important bug that we have missed for a long time.  I think that's unlikely.  But it does help mistakes as we write new opcodes.